doc/02-component-proof-of-concept/Cell.java

```diff
@@ -0,0 +1,105 @@
+import components.map.Map;
+import components.map.Map1L;
+
+/**
+ * Cell time.
+ *
+ * @author Elliott Simpson
+ */
+public final class Cell {
+
+    /**
+     * the map for relational data.
+     */
+    private Map relation;
+
+    /**
+     * the data stored in the cell node.
+     */
+    private T data;
+
+    /**
+     * does thing.
+     */
+    public Cell(T data) {
+        this.data = data;
+        this.creatRep();
+    }
+
+    /**
+     * Blank instantiation of cell.
+     */
+    private void creatRep() {
+        this.relation = new Map1L<T, Cell>();
+    }
+
+    // Kernel Methods
+
+    /**
+     * Checks that there are things related to this node.
+     *
+     * @return true if stuff in relation map, false if nothing
+     */
+    public boolean hasRelation() {
+        return this.relation.size() > 0;
+    }
+
+    /**
+     * Adds this cell to the relation map.
+     *
+     * @param item
+     *            an existing cell that needs to be added to the relation.
+     */
+    public void addCell(Cell item) {
+        this.relation.add(item.getData(), item);
+    }
+
+    /**
+     * Removes a specific cell from the relation map if it exists.
+     *
+     * @param item
+     *            the cell to remove
+     */
+    public void removeCell(Cell item) {
+        this.relation.remove(item.getData());
+    }
+
+    /**
+     * returns the map of relations. Might make this returned value immutable if possible so it doesn't mess with all other cells.
```

Comment:
Excellent point! I always recommend against returning underlying data to the user. They probably shouldn't know there is a map.